src/sgp/LoadSaveData.h

```diff
@@ -120,9 +121,10 @@ class DataReader
 	   encoded strings.	*/
 	auto readString(size_t const numChars, bool const stracLinuxFormat)
 	{
-		return std::invoke(
-			stracLinuxFormat ? &DataReader::readUTF32 : &DataReader::readUTF16,
-			this, numChars, ST_DEFAULT_VALIDATION);
+		if (stracLinuxFormat) {
+			return std::invoke(&DataReader::readUTF32, this, numChars, ST_DEFAULT_VALIDATION);
+		}
+		return std::invoke(&DataReader::readUTF16, this, numChars, true, ST_DEFAULT_VALIDATION);
```

Comment:
You can just call readUTF32 and readUTF16 directly; I used invoke() here only because the C++ syntax to make a call through a pointer to a member function is so ugly that I can never remember it.